Implement Login

thanghq-zinza opened the pull request:
Add Login page UI
Validate form and control Submit button
Implement fake API Call Result for testing without Server

thanghq-zinza committed:
Add Material UI + Redux Toolkit + React Router

thanghq-zinza committed:
switch to NextJS

thanghq-zinza committed:
Add prettier, vscode, editorconfig config

thanghq-zinza committed:
feat(ui): add login page

thanghq-zinza committed:
feat(ui): add login page

thanghq-zinza committed:
Merge branch 'implement_login' of https://github.com/zinza-academy/covid-19-vaccine-portal-thanghq24 into implement_login

thanghq-zinza committed:
feat(ui): Implement login page

Add full UI, basic routing to /register and /forgot-password
Implement validation, fake API call

hungnp194 commented on src/app/(authPage)/layout.tsx:
```diff
@@ -0,0 +1,39 @@
+  children
+}: {
+  children: React.ReactNode;
+}) {
```
Với các component có children thì dùng type : FC<PropsWithChildren> 
Với các component ko có children thì dùng type FC

hungnp194 commented on src/app/(authPage)/layout.tsx:
```diff
@@ -0,0 +1,39 @@
+import { Box, Stack } from '@mui/material';
+import Image from 'next/image';
+import React from 'react';
+import authImage from '../../../public/auth-page-img.png';
```
Dùng alias path. Tìm hiểu cách config compilerOptions.paths trong file tsconfig.json

hungnp194 commented on src/app/(authPage)/login/page.tsx:
```diff
@@ -0,0 +1,151 @@
+  email: string;
+  password: string;
+  loading: boolean;
+  success: boolean;
```
2 cái thuộc tính này ko thuộc LoginFormData.

Define type phải đảm bảo nó chỉ thực hiện đúng nhiệm vụ của nó

hungnp194 commented on src/app/(authPage)/login/page.tsx:
```diff
@@ -0,0 +1,151 @@
+      email: '',
+      password: '',
+      loading: false,
+      success: true
```
ko cho 2 cái này vào form

hungnp194 reviewed (changes requested):
feedback

thanghq-zinza committed:
implement alias paths

thanghq-zinza committed:
fix(ui): Change login form data

Make loading and success state as component state instead of form data
Clear some un-use imports, logs

thanghq-zinza committed:
refactor: Type all components to FC<>

thanghq-zinza committed:
fix: Format component type
